Presentation.Web/Controllers/API/V2/Internal/Excel/ExcelInternalV2Controller.cs

```diff
@@ -0,0 +1,213 @@
+        {
+            _excelService = excelService;
+            _excelApplicationService = excelApplicationService;
+        }
+
+        #region Excel Users
+
+        [HttpGet]
+        [Route("users")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public IHttpActionResult GetUsers(Guid organizationUuid, bool? exportUsers)
+        {
+            return _excelApplicationService.GetUsers(organizationUuid, exportUsers)
+                .Match(result => GetResponseMessage(result.MemoryStream, result.FileName), FromOperationError);
+        }
+
+        [HttpPost]
+        [Route("users")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        [SwaggerResponse(HttpStatusCode.Conflict)]
+        public async Task<IHttpActionResult> PostUsers(Guid organizationUuid, bool? importUsers)
+        {
+            var result = _excelApplicationService.ResolveOrganizationIdAndValidateAccess(organizationUuid);
+            if (result.Failed)
+                return FromOperationError(result.Error);
+
+            var organizationId = result.Value;
+            // check if the request contains multipart/form-data.
+            if (!Request.Content.IsMimeMultipartContent())
+                throw new HttpResponseException(HttpStatusCode.UnsupportedMediaType);
+
+            // read multipart form data
+            var stream = await ReadMultipartRequestAsync();
+
+            try
+            {
+                _excelService.ImportUsers(stream, organizationId);
+                return Ok();
+            }
+            catch (ExcelImportException e)
+            {
+                return Content(HttpStatusCode.Conflict, GetErrorMessages(e));
+            }
+        }
+
+        #endregion
+
+        #region Excel OrganizationUnits
+
+        [HttpGet]
+        [Route("units")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public IHttpActionResult GetOrgUnits(Guid organizationUuid, bool? exportOrgUnits)
+        {
+            return _excelApplicationService.GetUsers(organizationUuid, exportOrgUnits)
+                .Match(result => GetResponseMessage(result.MemoryStream, result.FileName), FromOperationError);
+        }
+
+        [HttpPost]
+        [Route("units")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        [SwaggerResponse(HttpStatusCode.Conflict)]
+        public async Task<IHttpActionResult> PostOrgUnits(Guid organizationUuid, bool? importOrgUnits)
+        {
+            var result = _excelApplicationService.ResolveOrganizationIdAndValidateAccess(organizationUuid);
```

Comment:
Consider extracting lines 99-110 (duplicated in all endpoints) to a helper method that returns Result<MemoryStream, operationError>